localstack/services/events/target.py

```diff
@@ -0,0 +1,304 @@
+from __future__ import annotations
+
+import json
+import logging
+import uuid
+from abc import ABC, abstractmethod
+
+from localstack.aws.api.events import (
+    Arn,
+    Target,
+    TargetId,
+)
+from localstack.aws.connect import ServiceLevelClientFactory, connect_to
+from localstack.utils import collections
+from localstack.utils.aws.arns import (
+    firehose_name,
+    parse_arn,
+    sqs_queue_url_for_arn,
+)
+from localstack.utils.aws.client_types import ServicePrincipal
+from localstack.utils.strings import to_bytes
+from localstack.utils.time import now_utc
+
+LOG = logging.getLogger(__name__)
+
+
+def connect_with_role(role_arn: str, region_name: str) -> ServiceLevelClientFactory:
+    """Connects using an assumed role."""
+    try:
+        return connect_to.with_assumed_role(
+            role_arn=role_arn, service_principal=ServicePrincipal.events, region_name=region_name
+        )
+    except ValueError as e:
+        LOG.debug(f"Could not connect with assumed role {role_arn}. Error: {e}")
```

Comment:
Does the fallback to account_id make sense here and is how AWS behaves?
Wouldn't a wrong `role_arn` result in an error?

Reply:
I think this needs to be tested in the future especially in regards to IAM policy streaming, since only role arn is required for permission to access EC2 instances, Kinesis Data Streams, Step Functions state machines and API Gateway APIs.

Reply:
In general, services get either an role passed, or use their service principal for these kind of requests. Some also support both, if you do not supply a role, it will use the service principal.
I am not sure what events supports, and it might differ by target, but we should definitely check that and only allow the right method.